native/cocos/renderer/gfx-gles3/GLES3GPUContext.cpp

```diff
@@ -319,13 +319,17 @@ void GLES3GPUContext::present(const GLES3GPUSwapchain *swapchain) {
         }
     }
 #endif
+    if (swapchain->eglSurface == EGL_NO_SURFACE) {
+        return;
+    }
```

Comment:
Will `swapchain->eglSurface` become `EGL_NO_SURFACE` in real applications?
If it is a pre-condition, consider using `CC_ASSERT(swapchain->eglSurface != EGL_NO_SURFACE)`.

Reply:
Yes, for an example, 2 windows changed to background will cause the `eglSurface `of both destroyed, and then make one of them foreground, and the other window's `eglSurface` will stays `EGL_NO_SURFACE`. But in `GLES3Device::present` it iterates all swapchains, and now the second window containing the invalid surface exists.

Reply:
We may add some comments here, to help understanding these problems.
Future maintainers might not think of this scenario.

Reply:
OK